From ae3540d2387f24f3620ac1cd03dc952df654264b Mon Sep 17 00:00:00 2001
From: Alan Jenkins <alan.christopher.jenkins@gmail.com>
Date: Mon, 25 Sep 2017 12:10:38 +0100
Subject: [PATCH] units: add missing Before=shutdown.target for units which it
 Conflicts

There's a few services missing this ordering.

Also remove a duplicate Conflicts=shutdown.target from
systemd-volatile-root.service.

(cherry picked from commit 0b9ad5bf2ec09e963f706fdba498d99f29f97a54)
---
 units/system-update-cleanup.service.in        | 1 +
 units/systemd-networkd-wait-online.service.in | 2 +-
 units/systemd-volatile-root.service.in        | 1 -
 3 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/units/system-update-cleanup.service.in b/units/system-update-cleanup.service.in
index 116be8bc2d..dc524da7a3 100644
--- a/units/system-update-cleanup.service.in
+++ b/units/system-update-cleanup.service.in
@@ -11,6 +11,7 @@ Documentation=man:systemd.special(5) man:systemd.offline-updates(7)
 After=system-update.target
 DefaultDependencies=no
 Conflicts=shutdown.target
+Before=shutdown.target
 
 # system-update-generator uses laccess("/system-update"), while a plain
 # ConditionPathExists=/system-update uses access("/system-update"), so
diff --git a/units/systemd-networkd-wait-online.service.in b/units/systemd-networkd-wait-online.service.in
index a84e91906d..89ca865b55 100644
--- a/units/systemd-networkd-wait-online.service.in
+++ b/units/systemd-networkd-wait-online.service.in
@@ -12,7 +12,7 @@ DefaultDependencies=no
 Conflicts=shutdown.target
 Requires=systemd-networkd.service
 After=systemd-networkd.service
-Before=network-online.target
+Before=network-online.target shutdown.target
 
 [Service]
 Type=oneshot
diff --git a/units/systemd-volatile-root.service.in b/units/systemd-volatile-root.service.in
index cc4e604e4c..c5a4ca3c27 100644
--- a/units/systemd-volatile-root.service.in
+++ b/units/systemd-volatile-root.service.in
@@ -12,7 +12,6 @@ DefaultDependencies=no
 Conflicts=shutdown.target
 After=sysroot.mount
 Before=initrd-root-fs.target shutdown.target
-Conflicts=shutdown.target
 AssertPathExists=/etc/initrd-release
 
 [Service]
